dotCMS/src/main/java/com/dotcms/contenttype/business/ContentTypeFactoryImpl.java

```diff
@@ -686,25 +718,42 @@ private void deleteContentletsByType(ContentType type) throws DotDataException {
   }
 
   private void deleteRelationships(ContentType type) throws DotDataException {
-    List<Relationship> relationships;
+
     //Deletes the child relationship field (if exists) if the parent is deleted.
-    relationships = APILocator.getRelationshipAPI().byParent(type);
-    for (final Relationship rel : relationships) {
+      final RelationshipAPI relationshipAPI = APILocator.getRelationshipAPI();
+      final List<Relationship> childRelationships = relationshipAPI.byParent(type);
+      final FieldAPI contentTypeFieldAPI = APILocator.getContentTypeFieldAPI();
+      for (final Relationship rel : childRelationships) {
       if(UtilMethods.isSet(rel.getParentRelationName()) && rel.isRelationshipField()) {
-        final Field fieldToDelete = APILocator.getContentTypeFieldAPI().byContentTypeIdAndVar(rel.getChildStructureInode(), rel.getParentRelationName());
-        APILocator.getContentTypeFieldAPI().delete(fieldToDelete);
+        final Field fieldToDelete = contentTypeFieldAPI
+                .byContentTypeIdAndVar(rel.getChildStructureInode(), rel.getParentRelationName());
+        contentTypeFieldAPI.delete(fieldToDelete);
       }
-      APILocator.getRelationshipAPI().delete(rel);
+      relationshipAPI.delete(rel);
     }
 
     //Deletes the parent relationship field if the child is deleted.
-    relationships = APILocator.getRelationshipAPI().byChild(type);
-    for (final Relationship rel : relationships) {
+    final List<Relationship> parentRelationships = relationshipAPI.byChild(type);
+    for (final Relationship rel : parentRelationships) {
       if(UtilMethods.isSet(rel.getChildRelationName()) && rel.isRelationshipField()) {
-        final Field fieldToDelete = APILocator.getContentTypeFieldAPI().byContentTypeIdAndVar(rel.getParentStructureInode(), rel.getChildRelationName());
-        APILocator.getContentTypeFieldAPI().delete(fieldToDelete);
+        final Field fieldToDelete = contentTypeFieldAPI
+                .byContentTypeIdAndVar(rel.getParentStructureInode(), rel.getChildRelationName());
+        contentTypeFieldAPI.delete(fieldToDelete);
       }
-      APILocator.getRelationshipAPI().delete(rel);
+      relationshipAPI.delete(rel);
+    }
+
+    //Once the parent - child : child - parent relationship has been cleared
+    //Now we need to remove the field from the actual CT.
+    final List<Field> fields = type.fields(RelationshipField.class);
```

Comment:
I am not sure this code fixes the root cause of the problem. I think it fixes the broken test, but not the problem in the existing logic

Reply:
I'll look into it once again. But I think it solves the problem of removing hanging relationship fields. The problem revealed by the test. The code isn't only meant to fix a test.

Reply:
The logic was faulty got it fixed Per @nollymar hunch